controllers/gitrepository_controller_test.go

```diff
@@ -220,7 +220,7 @@ var _ = Describe("GitRepositoryReconciler", func() {
 				reference:     &sourcev1.GitRepositoryRef{SemVer: "v1.0.0"},
 				waitForReason: sourcev1.GitOperationFailedReason,
 				expectStatus:  corev1.ConditionFalse,
-				expectMessage: "semver parse range error",
+				expectMessage: "no match found for semver: v1.0.0",
```

Comment:
The reason you had to change this is because it introduces a new bug, see: https://github.com/fluxcd/flux/issues/2729.

It can be fixed by using `semver.StrictNewVersion`, as described in: https://github.com/fluxcd/image-reflector-controller/issues/22#issuecomment-681928395, but this creates a new problem for Git repository tags as the prefix may have to be stripped from the tag first (and added back later).